src/PIL/TiffImagePlugin.py

```diff
@@ -1577,6 +1577,9 @@ def _save(im, fp, filename):
     # aim for 64 KB strips when using libtiff writer
     if libtiff:
         rows_per_strip = min((2 ** 16 + stride - 1) // stride, im.size[1])
+        # JPEG encoder expects multiple of 8 rows
+        if compression == "jpeg":
+            rows_per_strip = min(((rows_per_strip + 7) // 8) * 8, im.size[1])
```

Comment:
```suggestion
            rows_per_strip = min(((rows_per_strip + 7) // 8) * 8, rows_per_strip)
```
This seems better to me.